force-app/main/default/classes/GiftBatchForQueueable.cls

```diff
@@ -0,0 +1,148 @@
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of Salesforce.org nor the names of
+      its contributors may be used to endorse or promote products derived
+      from this software without specific prior written permission.
+
+    THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+    "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+    LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+    FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+    COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+    INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+    BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+    LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+    CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+    LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+    ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+    POSSIBILITY OF SUCH DAMAGE.
+*/
+/**
+* @author Salesforce.org
+* @date 2022
+* @description Gift Batch entity used when processing batches through BDI
+*/
+public inherited sharing class GiftBatchForQueueable {
+
+    private final Integer CHUNK_LIMIT = 50;
+
+    private GiftBatchId giftBatchId;
+    @TestVisible
+    private Gifts gifts;
+    @TestVisible
+    private List<List<Id>> chunkedIds = new List<List<Id>>();
+    @TestVisible
+    private GiftsSelectorForProcessing giftsSelector {
+        get {
+            if (giftsSelector == null) {
+                return new GiftsSelectorForProcessing();
+            }
+            return giftsSelector;
+        }
+        set;
+    }
+    @TestVisible
+    private Data_Import_Settings__c dataImportSettings {
+        get {
+            if (dataImportSettings == null) {
+                return BDI_DataImportService.loadSettings(giftBatchId.value());
+            }
+            return dataImportSettings;
+        }
+        set;
+    }
+    @TestVisible
+    private ElevateBatchService elevateBatchService {
+        get {
+            if (elevateBatchService == null) {
+                elevateBatchService = new ElevateBatchService();
+            }
+            return elevateBatchService;
+        }
+        set;
+    }
+    @TestVisible
+    private PS_IntegrationServiceConfig paymentIntegrationServiceConfig {
+        get {
+            if (paymentIntegrationServiceConfig == null) {
+                paymentIntegrationServiceConfig = new PS_IntegrationServiceConfig();
+            }
+            return paymentIntegrationServiceConfig;
+        }
+        set;
+    }
+
+    public GiftBatchForQueueable(GiftBatchId giftBatchId) {
+        this.giftBatchId = giftBatchId;
+    }
+
+    public GiftBatchId id() {
+        return giftBatchId;
+    }
+
+    public Boolean hasChunksToProcess() {
```

Comment:
There are a lot of public methods on this class.  Would it make sense to move any of them into the class where they are used and make them private?